### ISSUE TITLE
Detect name changes based on stored UUID

### ISSUE THREAD

**Issue by nothub** [minecraft-initiative/UniversalFriends#10]
problem example:

Alice friends Bob.
Bob changes name to PopBob.
Alice unfriends Bob, unaware he isnt called Bob anymore.

**Comment by IronException** [comment]
yea that's planned to happen

**Comment by 5HT2** [comment]
This is why you store as UUID and display as username

**Comment by IronException** [comment]
don't worry we only save UUIDs. I just haven't implemented it yet

**Comment by 5HT2** [comment]
^ ye so this can be closed when it is

**Comment by IronException** [comment]
^

**Comment by nothub** [comment]
the example i stated would require to save referr to old and new names, so alice is still able to remove bob even when he changed his name. otherwise alice wont find bob in her list anymore.

**Comment by nothub** [comment]
another solution would be to poll namemc api or smthg for the last change of the name when the change was older then a month and someone else registered the name.

**Comment by 5HT2** [comment]
That seems like it could break. How about having saving convert the username to UUID so you're always only storing UUID?

**Comment by nothub** [comment]
that would not meet the requirements of the problem i stated.

**Comment by 5HT2** [comment]
> This is why you store as UUID and display as username

Like what I said here?

**Comment by nothub** [comment]
yes, because when you just store the uuid and referr to the current username, you have no way for users to delete an entry by referring to the last username before a potential namechange.

**Comment by 5HT2** [comment]
No, they would have to refer to the new username, as would be listed in their friends list

**Comment by nothub** [comment]
yeah, and thats exactly the problem i see here. when you want to track your enemies, you wont be informed about their namechanges at all times.

**Comment by 5HT2** [comment]
You could store the UUID + current name and then check when changed, but that's up to the person using the library don't you think?

**Comment by nothub** [comment]
imho this should be a feature in the library, not smthg the user of the lib should have to implement.

**Comment by 5HT2** [comment]
So just a store + compare function? That would be a different issue to create then

**Comment by nothub** [comment]
yeah, my idea for this is either this:
> the example i stated would require to save referr to old and new names, so alice is still able to remove bob even when he changed his name. otherwise alice wont find bob in her list anymore.

or this:
> another solution would be to poll namemc api or smthg for the last change of the name when the change was older then a month and someone else registered the name.

**Comment by nothub** [comment]
ideally, this shouldnt rely on the namemc api. so i guess storing the last name locally would be the way to go.

**Comment by zeroeightysix** [comment]
Name change because the UUID thing has always been a thing. We do need something in place to see if usernames have changed however.
